### PR TITLE
Feature/opengeode geosciencesio

### PR DESCRIPTION
**Pull request by JulienChampagnol**

### CODE REVIEW

File: tests/test_functions.py [hunk 1]
```diff
@@ -22,16 +22,27 @@ def test_allowedfiles(client):
...
+        assert extension in extensions
 
 def test_allowedobjects(client):
-    # Normal test with filename
+    # Normal test with filename "corbi.og_brep"
     response = client.post(f"/{ID}/allowedobjects", data={"filename": "corbi.og_brep"})
     assert response.status_code == 200
     objects = response.json["objects"]
     assert type(objects) is list
+    list_objects = ["HybridSolid3D", "PolyhedralSolid3D", "TetrahedralSolid3D"]
```
Comment (panquez) [inline tests/test_functions.py:35]: ?

Comment (panquez) [reply tests/test_functions.py:35]: C'est plutot brep ?

**Review by panquez — COMMENTED** [review]
check test but ok

File: GeodeObjects.py [hunk 2]
```diff
@@ -1,20 +1,25 @@
...
-            "TriangulatedSurface2D": {"input": opengeode.TriangulatedSurfaceInputFactory2D, "output": opengeode.TriangulatedSurfaceOutputFactory2D, "load": opengeode.load_triangulated_surface2D, "save": opengeode.save_triangulated_surface2D},
-            "TriangulatedSurface3D": {"input": opengeode.TriangulatedSurfaceInputFactory3D, "output": opengeode.TriangulatedSurfaceOutputFactory3D, "load": opengeode.load_triangulated_surface3D, "save": opengeode.save_triangulated_surface3D},
-            "VertexSet": {"input": opengeode.VertexSetInputFactory, "output": opengeode.VertexSetOutputFactory, "load": opengeode.load_vertex_set, "save": opengeode.save_vertex_set}}
\ No newline at end of file
+    return {"BRep": {"input": [ opengeode.BRepInputFactory ], "output": [ opengeode.BRepOutputFactory ], "load": opengeode.load_brep, "save": opengeode.save_brep},
+            "CrossSection": {"input": [ opengeode_geosciences.CrossSectionInputFactory ], "output": [ opengeode_geosciences.CrossSectionOutputFactory ], "load": opengeode_geosciences.load_cross_section, "save": opengeode_geosciences.save_cross_section},
```
Comment (BotellaA) [inline GeodeObjects.py:8]:
```suggestion
            "CrossSection": {"input": [ opengeode_geosciences.CrossSectionInputFactory ], "output": [ opengeode.SectionOutputFactory, opengeode_geosciences.CrossSectionOutputFactory ], "load": opengeode_geosciences.load_cross_section, "save": opengeode_geosciences.save_cross_section},
```

File: blueprint.py [hunk 3]
```diff
@@ -77,39 +78,67 @@ async def convertfile():
         print("error : ", str(e))
         return {"status": 500, "error_message": str(e)}
 
-def ListObjects(ObjectsList, Extension):
+
+
+
```
Comment (BotellaA) [inline blueprint.py:83]:
```suggestion
```

File: tests/test_functions.py (hunk 1 above)
Comment (BotellaA) [reply tests/test_functions.py:35]: list_objects is not used. To remove

**Review by BotellaA — CHANGES_REQUESTED** [review]

File: tests/test_functions.py (hunk 1 above)
Comment (JulienChampagnol) [reply tests/test_functions.py]: Wrong copy paste, my bad you are wright

File: GeodeObjects.py (hunk 2 above)
Comment (JulienChampagnol) [reply GeodeObjects.py:8]: Does the CrossSection inherit from the Section ?

Comment (BotellaA) [reply GeodeObjects.py:8]: yes

**Review by BotellaA — APPROVED** [review]

**Pull request merged**